src/summary/LongitudinalTable.jsx

```diff
@@ -54,20 +56,26 @@ export default class LongitudinalTable extends Component {
             return 0;
         };
         dates.sort(sortFunction);
-        for (let dateIndex = 0; dateIndex < dates.length; dateIndex++) { //for each section, check to see if each date is in its list of dates.  If it is not, add an empty string to its array of data, at the dataIndex
+        dates.forEach((date, index) => { //if a data object doesn't have a date in its date list, add "" to the corresponding location in its data
             dataInfo.forEach((obj) => {
-                if (!Object.keys(obj.datesAndData).includes(dates[dateIndex])) {
-                    tableValues[dataInfo.indexOf(obj)].data.splice(dateIndex, 0, "");
+                if (!Object.keys(obj.datesAndData).includes(dates[index])) {
+                    tableValues[dataInfo.indexOf(obj)].data.splice(index, 0, "");
                 }
             });
-        }
-        return [tableValues, dates];
+        });
+        const starTableValues = [];
+        tableValues.forEach((section) => {
+            if (section.favorite) {
+                starTableValues.push(section);
+            }
+        });
+        return [tableValues, dates, starTableValues];
     }
     // updates current favorites and local storage
     toggleFavorites = (section) => {
         const newFavorites = this.state.favorites;
```

Comment:
We should probably be destructuring/cloning this object instead of just referencing it; since we splice it later, we'll be changing an object stored in state which we should avoid. Maybe this is a separate JIRA, since this isn't something directly introduced on this PR?

Reply:
would making it const newFavorites = [...this.state.favorites] fix the problem?

Reply:
I think so - assuming that favorites is an array, then that should work!

Reply:
resolved!